io-uring-test/src/tests/register_buf_ring.rs

```diff
@@ -580,6 +581,7 @@ where
     }
 
     let result = result as u32;
+    assert_eq!(result, len);
```

Comment:
When passing `0` as the length for the `opcode::Read` operation, this assertion fails with:
```
test register_buf_ring
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `44`', io-uring-test/src/tests/register_buf_ring.rs:584:5
```

Reply:
Isn't that a line you added?

Reply:
Yes, I added this assertion after a bit of drilling down on the original test failure (that one is part of the commit message now). This directly showed me that the problem was at the kernel boundary, as the read operation successfully completed with 0 bytes read.